samples/mpu/mem_domain_apis_test/src/main.c

```diff
@@ -103,21 +107,21 @@ void main(void)
 
 	/* create application threads */
 	k_thread_create(&app_thread_id[0], app_stack[0],
-			K_THREAD_STACK_SIZEOF(app_stack[0]), app_thread,
+			STACKSIZE, app_thread,
```

Comment:
why do this here? K_THREAD_STACK_SIZEOF() of the stack object in use should be sufficient.

Reply:
This was causing some issue on the MPU 1k alignment. This was leading to a stacking error.
So this stack sizeof returns a size - mpu guard size. Then inside the _new _thread API it will again subtract the size with the guard size. This was effectively making the stack size to become 0 in this case.

Reply:
Have we tried newer qemu, I think support for smaller than 1k size might be supported now.

Reply:
> Have we tried newer qemu, I think support for smaller than 1k size might be supported now.

Possibly in a 3.0.0-rc release.